apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -572,4 +574,63 @@ export class ApplicationController {
 
     return updatedApplication
   }
+
+  @Put('application/:id/requestFileSignature')
+  @ApiParam({
+    name: 'id',
+    type: String,
+    required: true,
+    description: 'The id of the application to update the state for.',
+    allowEmptyValue: false,
+  })
+  @ApiOkResponse({ type: ApplicationResponseDto })
+  @UseInterceptors(ApplicationSerializer)
+  async requestFileSignature(
+    @Param('id', new ParseUUIDPipe(), ApplicationByIdPipe)
+    application: Application,
+    @Body() input: RequestFileSignatureDto,
+  ): Promise<ApplicationResponseDto> {
+    const { type } = input
+
+    const {
+      controlCode,
+      documentToken,
+    } = await this.fileService.requestFileSignature(application, type)
+
+    const { updatedApplication } = await this.applicationService.update(
+      application.id,
+      {
+        attachments: {
+          ...application.attachments,
+          fileSignature: {
+            controlCode: controlCode,
+            documentToken: documentToken,
+          },
```

Comment:
For now, I would rather you store this under `application.externalData.fileSignature`.